src/daemon/osd_scenarios/osd_disk_activate.sh

```diff
@@ -77,10 +77,10 @@ function osd_activate {
   # - having the cleaning code just next to the concerned function in the same file is nice.
   function sigterm_cleanup_post {
     local ceph_mnt
-    ceph_mnt=$(findmnt --nofsroot --noheadings --output SOURCE --submounts --target /var/lib/ceph/osd/ | grep '^/')
+    ceph_mnt=$(findmnt --nofsroot --noheadings --output SOURCE --submounts --target "$(df --output=target | grep '/var/lib/ceph/osd/')")
```

Comment:
or maybe just use `df` for the mountpoint and use it on the `umount` command?

Reply:
I cannot really understand why you want to get the block device.
Why not having
for mnt in /var/lib/ceph/osd/*; do
umount mnt
done